ops/testing.py

```diff
@@ -1143,48 +1143,48 @@ def reset_planned_units(self):
         """
         self._backend._planned_units = None
 
-    def add_network(self, binding_name: str, address: str, *,
+    def add_network(self, binding_name: Optional[str], address: str, *,
```

Comment:
If the argument is optional, can it default to `None` like `relation_id`?

Reply:
I've considered this a bit more and talked with a couple more people, and decided that most charmers probably just want to set the default binding anyway, so I've made this a default arg now.

I've also renamed it to `endpoint`, which is what we call it in Juju these days. Either way it's non-ideal, because we use the "binding" terminology elsewhere in ops, but I think `endpoint` is best here.